### PR TITLE
CORE-5671: Ensure that all failures in the pipeline exception handling are caught

### PR DESCRIPTION
**Pull request by JamesHR3**
If the `FlowEventExceptionProcessor` is trying to handle a specific exception and another exception is thrown, this would previously result in the flow engine being taken down. This now escalates the error to the more generic `Exception` handling, which simply DLQs the flow. Additionally, this handling now applies to `Throwable`, which is slightly more generic.

I've also added a small log to some persistence code that helped me identify a problem on my side when setting up to do some verification of this.

### CODE REVIEW

File: components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/impl/FlowEventExceptionProcessorImpl.kt [hunk 1]
```diff
@@ -46,24 +46,24 @@ class FlowEventExceptionProcessorImpl @Activate constructor(
         maxRetryAttempts = config.getInt(PROCESSING_MAX_RETRY_ATTEMPTS)
     }
 
-    override fun process(exception: Exception): StateAndEventProcessor.Response<Checkpoint> {
-        log.error("Unexpected exception while processing flow, the flow will be sent to the DLQ", exception)
-        return StateAndEventProcessor.Response(
+    override fun process(throwable: Throwable): StateAndEventProcessor.Response<Checkpoint> = withEscalation {
+        log.error("Unexpected exception while processing flow, the flow will be sent to the DLQ", throwable)
```
Comment (lankydan) [inline components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/impl/FlowEventExceptionProcessorImpl.kt:50]: Just as an aside, why do we not post a new status when DLQing a flow? E.g. a DLQ'd status or something like that?

**Review by lankydan — DISMISSED** [review]
Left a comment but nothing needs to change here.

File: components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/impl/FlowEventExceptionProcessorImpl.kt (hunk 1 above)
Comment (JamesHR3) [reply components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/impl/FlowEventExceptionProcessorImpl.kt:50]: This is the handler of last resort. The assumption in this case is we might not be able to work out what the flow information is to be able to post a status event at all (e.g. the flow start context is corrupted), so we want to do the minimum possible that allows us to keep going.

For `FlowFatalException`s, we try and post a status update despite the DLQ.

File: components/virtual-node/virtual-node-write-service-impl/src/main/kotlin/net/corda/virtualnode/write/db/impl/writer/VirtualNodeWriterProcessor.kt [hunk 2]
```diff
@@ -258,6 +258,7 @@ internal class VirtualNodeWriterProcessor(
             try {
                 vaultDb.runCpiMigrations(dbChange)
             } catch (e: Exception) {
+                logger.error("Error running virtual node DB migration for CPI liquibase migrations: ${e.message}", e)
```
Comment (barrylapthorn) [inline components/virtual-node/virtual-node-write-service-impl/src/main/kotlin/net/corda/virtualnode/write/db/impl/writer/VirtualNodeWriterProcessor.kt]: Isn't the re-throw on the line below, 262, being logged as an error?

Comment (JamesHR3) [reply components/virtual-node/virtual-node-write-service-impl/src/main/kotlin/net/corda/virtualnode/write/db/impl/writer/VirtualNodeWriterProcessor.kt]: It wasn't appearing in the logs I was looking at earlier. I think it gets up to `handleException` on line 363, which then doesn't log, it just sets an error response to send back to the other side.

It's a fair point though, possibly the error should be logged there instead?

Comment (barrylapthorn) [reply components/virtual-node/virtual-node-write-service-impl/src/main/kotlin/net/corda/virtualnode/write/db/impl/writer/VirtualNodeWriterProcessor.kt]: Sounds like that might be better and probably fixes up some other places where exceptions aren't logged-to-error.

Comment (JamesHR3) [reply components/virtual-node/virtual-node-write-service-impl/src/main/kotlin/net/corda/virtualnode/write/db/impl/writer/VirtualNodeWriterProcessor.kt]: I've moved it - let me know what you think.

**Review by barrylapthorn — DISMISSED** [review]
lgtm (well, at least the vnode bit does)

**Review by lankydan — APPROVED** [review]

**Pull request merged**